internal/internal_workflow.go

```diff
@@ -1407,8 +1407,17 @@ func (s *selectorImpl) Select(ctx Context) {
 					if readyBranch != nil {
 						return false
 					}
-					readyBranch = func() {
+					// readyBranch is not executed when AddDefault is specified,
+					// setting the value here prevents the signal from being dropped
+					dropSignalFlag := getWorkflowEnvironment(ctx).TryUse(SDKFlagBlockedSelectorSignalReceive)
```

Comment:
Won't the `TryUse` here set the flag?

Reply:
It didn't seem like there was an existing method to query flags without setting them, created a new `GetFlag` method for this